Assets/Tests/Common/CodeCoverageHistoryCleaner.cs

```diff
@@ -0,0 +1,170 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+
+namespace PlayEveryWare.EpicOnlineServices.Tests
+{
+    using EpicOnlineServices.Editor.Utility;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Xml.Linq;
+
+    using System;
+    using System.Linq;
+    using UnityEditor;
+    using UnityEngine;
+    using Utility;
+    using JsonUtility = Utility.JsonUtility;
+
+    /// <summary>
+    /// This class consolidates any code coverage history data output from the
+    /// code coverage utility by reading all code coverage history files, and
+    /// keeping only the last code coverage report for each date.
+    /// </summary>
+    internal class CodeCoverageHistoryCleaner
+    {
+        /// <summary>
+        /// The directory (relative to the root of the repository) at which the
+        /// code coverage tool looks for historic coverage data.
+        /// </summary>
+        private const string CODE_COVERAGE_HISTORIC_DATA_DIRECTORY = "tools/coverage/historic";
+
+        /// <summary>
+        /// The format used by Unity's Code Coverage tool for recording the
+        /// date and time of a code coverage report.
+        /// </summary>
+        private const string CODE_COVERAGE_DATETIME_FORMAT = "yyyy-MM-dd_HH-mm-ss";
+
+        /// <summary>
+        /// The file name to use when consolidating the code coverage history.
+        /// </summary>
+        private const string CODE_COVERAGE_CONSOLIDATED_FILE = "Consolidated_CoverageHistory.xml";
+
+        /// <summary>
+        /// The file in the code coverage report from which to determine the
+        /// current percentage of lines of code covered by the unit tests. 
+        /// </summary>
+        private const string CODE_COVERAGE_REPORT_SUMMARY_FILE = "tools/coverage/results/Report/Summary.xml";
+
+        /// <summary>
+        /// The file in which to store the percentage of lines covered, to
+        /// subsequently be used to generate a code coverage badge for GitHub.
+        /// </summary>
+        private const string CODE_COVERAGE_PERCENTAGE_FILE = "tools/coverage/results/current_coverage.txt";
+
+        public static void ExtractLineCoverage()
+        {
+            XDocument summaryDoc = XDocument.Load(CODE_COVERAGE_REPORT_SUMMARY_FILE);
+
+            var summaryElement = summaryDoc.Descendants("Summary").First();
+
+            ulong coveredLines = ulong.Parse(summaryElement.Descendants("Coveredlines").First()?.Value);
+            ulong coverableLines = ulong.Parse(summaryElement.Descendants("Coverablelines").First()?.Value);
+
+            double percentage = (double)coveredLines / coverableLines * 100;
```

Comment:
Is there ever a world where coverableLines can be 0?

Reply:
Not really, but I added a check for that since doing so is good practice.